services/rollout-service/pkg/argo/argo_test.go

```diff
@@ -782,7 +788,201 @@ func TestArgoConsume(t *testing.T) {
+			ExpectedOperations: 0,
+		},
+		{
+			Name: "when filter has `*`",
+			Overview: &api.GetOverviewResponse{
+				Applications: map[string]*api.Application{
+					"foo": {
+						Releases: []*api.Release{
+							{
+								Version:        1,
+								SourceCommitId: "00001",
+							},
+						},
+						Team: "footeam",
+					},
+				},
+				EnvironmentGroups: []*api.EnvironmentGroup{
+					{
+
+						EnvironmentGroupName: "staging-group",
+						Environments: []*api.Environment{
+							{
+								Name: "staging",
+								Applications: map[string]*api.Environment_Application{
+									"foo": {
+										Name:    "foo",
+										Version: 1,
+										DeploymentMetaData: &api.Environment_Application_DeploymentMetaData{
+											DeployTime: "123456789",
+										},
+									},
+								},
+								Priority: api.Priority_UPSTREAM,
+								Config: &api.EnvironmentConfig{
+									Argocd: &api.EnvironmentConfig_ArgoCD{
+										Destination: &api.EnvironmentConfig_ArgoCD_Destination{
+											Name:   "staging",
+											Server: "test-server",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				GitRevision: "1234",
+			},
+			Application: &api.Environment_Application{
+				Name:    "foo",
+				Version: 1,
+				DeploymentMetaData: &api.Environment_Application_DeploymentMetaData{
+					DeployTime: "123456789",
+				},
+			},
+			Environment: &api.Environment{
+				Name: "development",
+				Applications: map[string]*api.Environment_Application{
+					"foo": {
+						Name:    "foo",
+						Version: 1,
+						DeploymentMetaData: &api.Environment_Application_DeploymentMetaData{
+							DeployTime: "123456789",
+						},
+					},
+				},
+				Priority: api.Priority_UPSTREAM,
+				Config: &api.EnvironmentConfig{
+					Argocd: &api.EnvironmentConfig_ArgoCD{
+						Destination: &api.EnvironmentConfig_ArgoCD_Destination{
+							Name:   "development",
+							Server: "test-server",
+						},
+					},
+				},
+			},
+			ArgoManageEnabled:  true,
+			ArgoManageFilter:   []string{"*"},
+			ExpectedOperations: 1,
+		},
+	}
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			as := &mockApplicationServiceClient{
+				cancel:    cancel,
+				t:         t,
+				lastEvent: make(chan *ArgoEvent, 10),
+			}
+			hlth := &setup.HealthServer{}
+			argoProcessor := &mockArgoProcessor{
+				lastOverview:          tc.Overview,
+				ApplicationClient:     as,
+				trigger:               make(chan *api.GetOverviewResponse, 10),
+				argoApps:              make(chan *v1alpha1.ApplicationWatchEvent, 10),
+				ManageArgoAppsEnabled: tc.ArgoManageEnabled,
+				ManageArgoAppsFilter:  tc.ArgoManageFilter,
+			}
+			hlth.BackOffFactory = func() backoff.BackOff { return backoff.NewConstantBackOff(0) }
+
+			argoProcessor.CreateOrUpdateApp(ctx, tc.Overview, tc.Application, tc.Environment, tc.AppsKnownToArgo)
```

Comment:
This test case only tests the `mockArgoProcessor`. Do we actually need it?

Reply:
I wanted to test the filter on the mockArgoProcessor. Hmm, I can see if I can rearrange the function call.

Reply:
I changed the test to only test the function regarding the selfManage filter (testing both the error output and the isActive flag).